### PR TITLE
Add version utilities

### PR DESCRIPTION
**Pull request by ernilambar**
Add version utilities:

- Add `get_wordpress_stable_version()` similar to that helper method in `Plugin_Readme_Check` class
- New methods to fetch version informations
   * get_wordpress_latest_version()
- Added `Version_Utils_Tests` unit test class

### CODE REVIEW

**Comment by ernilambar** [comment]
On second thought, this could be alternative naming convention:

- get_wordpress_stable_version()
- get_wordpress_latest_version()
- get_php_required_version()
- get_php_recommended_version()
- get_mysql_required_version()

File: includes/Traits/Version_Utils.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * Trait for version utilities.
+ *
+ * @since 1.4.0
+ */
+trait Version_Utils {
+
+	/**
+	 * Returns current major WordPress version.
+	 *
+	 * @since 1.0.0
```
Comment (mukeshpanchal27) [inline includes/Traits/Version_Utils.php]:
```suggestion
	 * @since 1.4.0
```

**Comment by ernilambar** [comment]
PR updated with new naming pattern.

Also added `get_wordpress_relative_major_version()` method to find relative WP major version.

File: includes/Traits/Version_Utils.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+					set_transient( 'wp_plugin_check_latest_version_info', $info, DAY_IN_SECONDS );
+				}
+			}
+		}
+
+		return array_key_exists( $key, $info ) ? $info[ $key ] : null;
+	}
+
+	/**
+	 * Returns the current WordPress' required PHP version.
```
Comment (swissspidy) [inline includes/Traits/Version_Utils.php]: Why do we need that? Same for the current site's required MySQL version

Comment (ernilambar) [reply includes/Traits/Version_Utils.php]: In our internal addon we are trying to use this like checking this value vs PHP version mentioned in the plugin header. We can remove get_mysql_required_version() as I dont see its implementation soon in the near future.

Comment (ernilambar) [reply includes/Traits/Version_Utils.php]: We could also use PHP version values in message `The "Requires PHP" header in the plugin file should only contain a PHP version such as "5.2.4" or "7.0".` where currently those versions are hardcoded.

Comment (swissspidy) [reply includes/Traits/Version_Utils.php]: Those being hardcoded seems fine to me

**Review by swissspidy — APPROVED** [review]

**Review by davidperezgar — APPROVED** [review]

**Pull request merged**